makePoplogCommander.sh

```diff
@@ -476,46 +664,76 @@ void extendPath( char * prefix, char * path, char * suffix ) {
+
+    int vflags = flags & VARIANT_FLAGS;
+    switch ( vflags ) {
 ****
-echo
 
-################################################################################
-# We now run the popenv.sh script inside a clean environment to capture the 
-# set of environment variables needed. We then need to replace any matches
-# of the string "_build/poplog_base" ($usepop) with our unique value USEPOP.
-# This will allow us to dynamically substitute with the selfHome'd value
-# at run-time. 
-#
-# This is inherently a weak strategy because it relies on being able to 
-# identify substitutions of $usepop. We improve its robustness by doing the
-# process twice with different values of $usepop - using the '..' trick.
-# If the resultant code is not identical we have a problem and we halt.
-# (N.B It is probably not necessary to run the variables through sort but I 
-# couldn't find a clear guarantee that env generates a sorted list.)
-################################################################################
+for variant in $VARIANTS
+do
+    echo "        case VARIANT_${variant^^}:"
+    echo "            ${variant}_setUpEnvVars( base, inherit_env );"
+    echo "            break;"
+done
 
-CODE1=`env -i sh -c '(usepop="_build/poplog_base" && . $usepop/pop/com/popenv.sh && env)' | sort \
-| grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' \
-| sed -e 's!_build/poplog_base![//USEPOP//]!g' \
-| sed -e 's/"/\\"/g' \
-| sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'`
+# We have to default to 'xt' at the moment because the 'xm' build has issues.
+cat << \****
+        default:
+            if ( inherit_env ) {
+                char * use_build = getenv( "POPLOG_USE_BUILD" );
+                if ( use_build == NULL ) {
+                    use_build = "xm"; 
+                }
+                if ( 0 ) {
+                    // Skip
+****
 
-CODE2=`env -i sh -c '(usepop="_build/poplog_base/pop/.." && . $usepop/pop/com/popenv.sh && env)' | sort \
-| grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' \
-| sed -e 's!_build/poplog_base/pop/..![//USEPOP//]!g' \
-| sed -e 's/"/\\"/g' \
-| sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'`
+for variant in $VARIANTS
+do
+    echo '                } else if ( strEquals( "'$variant'", use_build ) ) {'
+    echo "                    ${variant}_setUpEnvVars( base, inherit_env );"
+done
 
-if [ "$CODE1" != "$CODE2" ]; then
-    exit 1
-fi
+cat << \****
+                } else {
+                    xm_setUpEnvVars( base, inherit_env );
+                }
+            } else {
+                nox_setUpEnvVars( base, inherit_env );
+            }
+            break;
+    }
+
+****
+
+# Now we squirt in the shared environment variables.
+cat _build/environments/shared.env \
+| sed -e 's/"/\\"/g' \
+| sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'
```

Comment:
This code is duplicated and is finicky enough that it should be refactored out.

Reply:
Done.